fetch.js

```diff
@@ -246,14 +246,20 @@ function Body() {
       this._bodyText = body = Object.prototype.toString.call(body)
     }
 
-    if (!this.headers.get('content-type')) {
+    const contentType = this.headers.get('content-type')
+
+    if (!contentType) {
       if (typeof body === 'string') {
         this.headers.set('content-type', 'text/plain;charset=UTF-8')
       } else if (this._bodyBlob && this._bodyBlob.type) {
         this.headers.set('content-type', this._bodyBlob.type)
       } else if (support.searchParams && URLSearchParams.prototype.isPrototypeOf(body)) {
         this.headers.set('content-type', 'application/x-www-form-urlencoded;charset=UTF-8')
       }
+    } else if (contentType.includes('json') && typeof this._bodyInit !== 'string') {
```

Comment:
This isn't a reliable JSON request body content-type header check. Would you recommend to provide `application/json; charset=UTF-8` instead? 

I'm thinking that this behavior should also affect _any_ JSON-like content-type header (i.e. `application/hal+json`).

Reply:
- [ ]